web-app/src/main/webapp/script.js

```diff
@@ -70,17 +75,57 @@ function showReportForm() {
 }
 
 async function loadPoliceReports(map) {
+  // Clear all markers on the map
+  mapMarkers.map(marker => marker.setMap(null));
+  mapMarkers.length = 0;
+
   const FILE_NAMES = ['2019_12_london', '2020_01_london', '2020_02_london', '2020_03_london', '2020_04_london', '2020_05_london']
+  const uncheckedCategoriesElement = Array.from(document.querySelectorAll("input.category:not(:checked)"));
+  const uncheckedCategories = uncheckedCategoriesElement.map(element => element.value);
+
   for (const file_name of FILE_NAMES) {
+    if (!withinTimeFrame(file_name)) {
+      continue;
+    }
     const data = await fetch('../data/' + file_name + '.json');
     const reports = await data.json();
-    for (report of reports) {
-      new google.maps.Marker({
-        position: {
-          lat: Number(report.latitude),
-          lng: Number(report.longitude)
-        }, map: map
-      });
-    };
+
+    const filteredReports = reports.filter(report => {
+      if (report.latitude == "" || report.longitude == "") {
+        return false;
+      }
+      return displayCrimeType(uncheckedCategories, report.crimeType);
+    });
+
+    const markers = filteredReports.map(report => new google.maps.Marker({
+      position: {
+        lat: Number(report.latitude),
+        lng: Number(report.longitude)
+      }, map: map
+    }));
+
+    mapMarkers = mapMarkers.concat(markers);
+  }
+}
+
+function displayCrimeType(uncheckedCategories, crimeType) {
+  // Don't display if category unchecked
+  for (category of uncheckedCategories) {
+    if (crimeType.toLowerCase().includes(category)) {
+      return false;
+    }
   }
```

Comment:
Potentially something to think about for the future: since our crime categories don't change, it might be easier to just assign each category a number when handling it in the backend?